src/components/ui/Box.tsx

```diff
@@ -68,10 +83,46 @@ export const Box: React.FC<BoxProps> = ({
     ? longPressEvents
     : { onClick: canClick ? onClick : undefined };
 
+  useEffect(() => {
+    if (isHover) {
```

Comment:
This is some really nested / hard to read code. Do you think we could refactor a little and maybe have some returns to break out of the nested condtionals?

`labelChecker` is also a difficult to understand var name.

Reply:
I refactor the code and came up with a better name for the label checker.  It should look way better now 😊